source/concurrent/RecycleBin.ooc

```diff
@@ -14,48 +14,39 @@
 * You should have received a copy of the GNU Lesser General Public License
 * along with This software. If not, see <http://www.gnu.org/licenses/>.
 */
-import threading/Mutex
+use concurrent
 
 RecycleBin: class <T> {
-	_mutex := Mutex new()
 	_free: Func (T)
-	//TODO: Replace with SynchronizedVectorList
-	_list: VectorList<T>
+	_list: SynchronizedList<T>
 	_size: Int
 	count ::= this _list count
 	isFull ::= this count == this _size
 	isEmpty ::= this _list empty
-	init: func (=_size, =_free) { this _list = VectorList<T> new(_size) }
+	init: func (=_size, =_free) { this _list = SynchronizedList<T> new(_size) }
 	free: override func {
 		this clear()
 		this _list free()
-		this _mutex free()
 		super()
 	}
 	add: func (object: T) {
-		this _mutex lock()
 		if (this isFull)
 			this _free(this _list remove(0))
 		this _list add(object)
-		this _mutex unlock()
 	}
 	search: func (matches: Func (T) -> Bool) -> T {
 		result: T = null
-		this _mutex lock()
 		for (i in 0 .. this _list count) {
 			if (matches(this _list[i])) {
```

Comment:
I just realized this will now be much more expensive since the mutex lock in the list will be locked/unlocked twice for every element in the list (`this _list[i]` and `this _list remove(i)`). Previously it was only locked once before and after the loop. Can you use the `apply` function and solve this with a single lock/unlock like before?